### ISSUE TITLE
[Suggestion] Improve the fully navigable menu and removing "Top"

### ISSUE THREAD

**Issue by aphillipo** [elixir-lang/ex_doc#650]
I love that ExDoc has a fully navigable menu structure on every page without reload, but I strongly dislike "Top" to actually get to the page (because it's meaningless and repetitive - it might as well say "go" or "start", all of which feel weird).  It really wasn't clear to me what was happening the first time I used it and I've been thinking about a better UI for this for a while.

Let's see if you guys like this proposal:

1) The current pages' module/navbar item is always open.  Clicking the name goes to the top of the current document, maybe with an up arrow next to it.

2) Clicking the heading of a top level navbar item (that is not the current page) opens the menu with subsections/functions as now but no "Top".  Instead two new icons appear; a minus/collapse icon on the left and an arrow on the far right.  

3) Clicking on the open menu item heading (and arrow) does not close the section but takes you to the "Top".  Clicking the minus closes the open menu.

4) Optionally, clicking another menu closes the open menu and does step 2 (the current pages menu is always open so if you want to browse you can click twice on a heading and then open another submenu; 3 open function definitions is probably overkill anyway).

5) Double clicking should cancel the opening of the menu and open the page.

Maybe thinking about 5) with a small delay would be enough; I still would want to get rid of "Top" because I found it so unclear that this means "Navigate to the document".

Apologies if you absolutely hate this or this is the wrong place, but I think it would work quite well ;-)

I'm happy to have a look at implementing too if you like!  Ta.

**Comment by josevalim** [comment]
Can you do a quick mock up of what you have in mind? At least visually? It has been very hard to put the pieces together only with text.

**Comment by aphillipo** [comment]
<img width="317" alt="plug-ex_doc_screenshot" src="https://cloud.githubusercontent.com/assets/52347/20772631/2e3ed5be-b746-11e6-8155-49fe98f641b1.png">

**Comment by josevalim** [comment]
To be very honest I am not convinced about the proposed changes. I would have no idea when to click on `[+]` when to click on `=>` and when to click on `+`. Ideally we would use the same `+` we use for "Functions" and add a distinction between clicking on the name and on the "+" sign. It leads to less clicks but I am not sure if it is less confusing.

**Comment by aphillipo** [comment]
Fair enough, I'll close this then, I'm still not convinced "Top" is the right messaging; I guess I'm trying to hack around this.  Thanks for taking a look.

**Comment by eksperimental** [comment]
the only way this wouldn't be confusing would be opening the tree branches with the +/- icon. and clicking on the module names and functions takes you to that page.

**Comment by aphillipo** [comment]
It's a very small clickable area tho... not too sure.  Like I say I find anything that get's rid of "Top" to be preferable; I can't put my finger on why I dislike it so much but it's just me probably!

**Comment by wsmoak** [comment]
It is not just you @aphillipo . :)  I commented on usability the first time I saw it, but I do not have the expertise to try to fix it.  Anecdotally, I have been all over the docs looking up things for Advent of Code, and I _still_ stare at the screen waiting for the content for the menu item I just clicked on to appear.  Then I have to look left again and find that item and the "Top" underneath it (since more than one may be expanded) in order to see the docs.  If I know which one I want, I find it's quicker to just edit the URL.
